webapp/channels/src/components/common/hooks/useAccessControlAttributes.ts

```diff
@@ -0,0 +1,158 @@
+
+/**
+ * A hook for fetching access control attributes for an entity
+ *
+ * @param entityType - The type of entity (e.g., 'channel')
+ * @param entityId - The ID of the entity
+ * @param hasAccessControl - Whether the entity has access control enabled
+ * @returns An object containing the attribute tags, loading state, and fetch function
+ */
+export const useAccessControlAttributes = (
+    entityType: EntityType,
+    entityId: string | undefined,
+    hasAccessControl: boolean | undefined,
+) => {
+    const [attributeTags, setAttributeTags] = useState<string[]>([]);
+    const [structuredAttributes, setStructuredAttributes] = useState<AccessControlAttribute[]>([]);
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState<Error | null>(null);
+    const dispatch = useDispatch();
+
+    // Helper function to process attribute data and extract tags
+    const processAttributeData = useCallback((data: Record<string, string[]> | undefined) => {
+        if (!data) {
+            setAttributeTags([]);
+            setStructuredAttributes([]);
+            return;
+        }
+
+        const tags: string[] = [];
+        const attributes: AccessControlAttribute[] = [];
+
+        // Extract values from all properties in the response
+        // Format: { "attributeName": ["value1", "value2"], "anotherAttribute": ["value3"] }
+        Object.entries(data).forEach(([name, values]) => {
+            // Add to structured format
+            if (Array.isArray(values)) {
+                attributes.push({name, values: [...values]});
+
+                // Add to flat tags (existing behavior)
+                values.forEach((value) => {
+                    if (value !== undefined && value !== null) {
+                        tags.push(value);
+                    }
+                });
+            }
+        });
+
+        setAttributeTags(tags);
+        setStructuredAttributes(attributes);
+    }, []);
+
+    const fetchAttributes = useCallback(async (forceRefresh = false) => {
+        if (!entityId || !hasAccessControl) {
+            return;
+        }
+
+        // Set loading state at the beginning
+        setLoading(true);
+        setError(null);
+
+        try {
+            // Validate entity type first
+            if (!SUPPORTED_ENTITY_TYPES.includes(entityType)) {
+                throw new Error(`Unsupported entity type: ${entityType}`);
+            }
+
+            // Check cache first (unless forceRefresh is true)
+            const cacheKey = `${entityType}:${entityId}`;
+            const cachedEntry = attributesCache[cacheKey];
+            const now = Date.now();
+
+            // Use cache if it exists and is not too old and forceRefresh is false
+            // But still set loading to false to trigger a state update for tests
+            if (!forceRefresh && cachedEntry && (now - cachedEntry.timestamp < CACHE_TTL)) {
+                processAttributeData(cachedEntry.data);
```

Comment:
Even on cache hit, we'd still iterate over the data in `processAttributeData` function, , generate two new arrays and do some computation. Why can't jus simply cache the result of `processAttributeData` function instead and just return the cached data and do absolutely nothing in this hook on cache hit?

Reply:
yes, totally agree, will be part of my next commit.